src/components/Button.js

```diff
@@ -0,0 +1,24 @@
+import { func } from 'prop-types';
+import { string } from 'prop-types';
+
+function Button(props) {
+  return (
+    <button
+      className={'button button--' + (props.size ? props.size : 'medium')}
+      onClick={props.onClick}
+    >
+      {props.iconOnly ? (
+        <img alt={props.children} src={require('../images/' + props.icon)} />
```

Comment:
Don't use `require` in your code - CRA projects use `import`s by default and we should stick to one solution.

React has a built-in mechanism that allows you to import `.svg` files and use them as regular components:

```jsx
// App.jsx
import { ReactComponent as IconName } from '../images/icon.svg';

const App = () => {
    render (
        <IconName />
    );
}
```

You can then pass the component as a prop:

```jsx
// App.jsx
import { ReactComponent as IconName } from '../images/icon.svg';

const App = () => {
    render (
        <button icon={IconName} ... />
    );
}

// Button.jsx
const Button = ({ icon: IconComponent }) => {
    return (
        <button>
            <IconComponent />
        </button>
    );
}
```